pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -329,35 +359,124 @@ func (w *Webhook) getAllLatestLibraries() []libInfo {
 	return libsToInject
 }
 
-// extractLibInfo returns the language, the image,
-// and a boolean indicating whether the library should be injected into the pod
-func (w *Webhook) extractLibInfo(pod *corev1.Pod) ([]libInfo, bool) {
-	// If the pod is "injectable" and annotated with libraries to inject, use those.
-	if w.isPodEligible(pod) {
-		// The library version specified via annotation on the Pod takes precedence
-		// over libraries injected with Single Step Instrumentation
-		libs := w.extractLibrariesFromAnnotations(pod)
-		if len(libs) > 0 {
-			return libs, false
-		}
+// libInfoSource describes where we got the libraries from for
+// injection and is used to set up metrics/telemetry. See
+// Webhook.injectAutoInstruConfig for usage.
+type libInfoSource int
+
+const (
+	// libInfoSourceNone is no source provided.
+	libInfoSourceNone libInfoSource = iota
+	// libInfoSourceLibInjection is when the user sets up manual annotations on their pods.
+	libInfoSourceLibInjection
+	// libInfoSourceSingleStepInstrumentation is when we are using the instrumentation config
+	// to determine which libraries to inject.
+	libInfoSourceSingleStepInstrumentation
+	// libInfoSourceSingleStepLanguageDetection is when we use the language detection
+	// annotation to determine which libs to inject.
+	libInfoSourceSingleStepLangaugeDetection
```

Comment:
Mentioning just in case, 

Even when using language detection to inject libraries, we still depend on the config of the single step instrumentation. So, there is an overlap between these 2. 

Ideally, when everything is stable, I think language detection should become an implementation detail of SSI, and shouldn't be perceived as a separate source. 

For now, it might be okay to make this distinction.

Reply:
Agree- it’s there right now only to disambiguate some metrics we are producing.